src/typescript-worker/typescript.ts

```diff
@@ -109,12 +118,22 @@ export class TypeScriptBuilder {
 
     // Wait for all typings to be fetched, and then retrieve slower semantic
     // diagnostics.
-    for (const [specifier, content] of await typesFetcher.getFiles()) {
+    const typesFetcher = new TypesFetcher(
+      this._cdn,
+      this._importMapResolver,
+      packageJson
+    );
```

Comment:
Does the `TypesFetcher` do any caching of files already loaded?

Reply:
We cache results by specifier (package + version + path) *within* each build to prevent duplicate fetches/analysis, but yeah -- there is currently no caching *across* builds. This is definitely something to address in a follow up. If the imports and package.json haven't changed since last build, then we can re-use all of this work, whereas currently we recompute all of it. Note that the underlying browser HTTP cache will take care of the main network cost of this duplicated work, but there's also a lot of parsing and analysis work being done that probably adds noticeable latency.

Filed https://github.com/PolymerLabs/playground-elements/issues/185.